src/renderers/shared/fiber/ReactFiberCommitWork.js

```diff
@@ -86,7 +86,11 @@ module.exports = function<T, P, I, TI, C, CX>(
   function attachRef(current : ?Fiber, finishedWork : Fiber, instance : any) {
     const ref = finishedWork.ref;
     if (ref && (!current || current.ref !== ref)) {
-      ref(instance);
+      if (typeof config.getPublicInstance === 'function') {
```

Comment:
Please destructure this from config earlier. We don't want to read that value every time. I think we could also make this required to keep contract strict.